extensions/amp-access-fewcents/0.1/fewcents-impl.js

```diff
@@ -0,0 +1,183 @@
+   */
+  constructor(accessService, accessSource) {
+    /** @const */
+    this.ampdoc = accessService.ampdoc;
+
+    /** @const @private {!../../amp-access/0.1/amp-access-source.AccessSource} */
+    this.accessSource_ = accessSource;
+
+    /** @private {?Node} */
+    this.innerContainer_ = null;
+
+    /** @private {?Node} Main container where paywall will be rendered */
+    this.dialogContainer_ = null;
+
+    /** @const @private {JsonObject}  Stores the config passed by the publisher */
+    this.fewcentsConfig_ = this.accessSource_.getAdapterConfig();
+
+    /** @private {string} Authorize endpoint to get paywall data*/
+    this.authorizeUrl_ = this.prepareAuthorizeUrl_();
+
+    /** @private {!JsonObject} */
+    this.i18n_ = Object.assign(dict(), DEFAULT_MESSAGES);
+
+    // Install styles.
+    installStylesForDoc(this.ampdoc, CSS, () => {}, false, TAG);
+  }
+
+  /**
+   * Decides to show the paywall or publisher content by calling authorize endpoint
+   * @return {!Promise<!JsonObject>}
+   */
+  authorize() {
+    return this.getPaywallData_().then(
+      // [TODO] : More edge cases to be added with actual call
+      (response) => {
+        // Removing the paywall from dialogContainer
+        this.emptyContainer_();
+        // Showing the publisher's content
+        return {access: response.data.access};
+      },
+      (err) => {
+        // Rendering the paywall when request promise is rejected
+        const {response} = err;
+        return response.json().then(() => {
+          this.emptyContainer_().then(this.renderPurchaseOverlay_.bind(this));
+          return {access: false};
+        });
+      }
+    );
+  }
+
+  /**
+   * Add request parameters for the authorize endpoint
+   * @return {string}
+   * @private
+   */
+  prepareAuthorizeUrl_() {
+    // [TODO]: Actual implementation
+    dev().fine(TAG, 'Publishers config', this.fewcentsConfig_);
+    return;
+  }
+
+  /**
+   * Get paywall data by calling authorize endpoint
+   * @return {!Promise<Object>}
+   * @private
+   */
+  getPaywallData_() {
+    // [TODO]: Actual API implementation using fetch
+    dev().fine(TAG, 'authorizeUrl', this.authorizeUrl_, CONFIG_BASE_PATH);
+
+    // Currently, case only for showing the paywall later in future depend upon API response
+    return Promise.reject({
+      response: {
+        status: 402,
+        json() {
+          return Promise.resolve({success: true});
+        },
+      },
+    });
+  }
+
+  /**
+   * Removes the paywall from the element on publisher's page where paywall is displayed
+   * @private
+   * @return {!Promise}
+   */
+  emptyContainer_() {
+    // [TODO]: Actual implementation
+    return Promise.resolve();
+  }
+
+  /**
+   * Creates new element for the paywall
+   * @param {string} name
+   * @return {!Element}
+   * @private
+   */
+  createElement_(name) {
+    return this.ampdoc.win.document.createElement(name);
```

Comment:
nit: we have a few helpers for creating elements in `src/core/dom/index.js` you can use when appropriate.

Reply:
I will have a look at them, and update it.

Reply:
Changes done.